pkg/executor/fscache/poolcache.go

```diff
@@ -138,13 +142,21 @@ func (c *PoolCache) service() {
 					break
 				}
 			}
+
 			if found {
+				// we found a specialized pod with room for another request, use it
 				req.responseChannel <- resp
 				continue
 			}
-			specializationInProgress := funcSvcGroup.svcWaiting - funcSvcGroup.queue.Len()
-			capacity := ((specializationInProgress + len(funcSvcGroup.svcs)) * req.requestsPerPod) - (totalActiveRequests + funcSvcGroup.svcWaiting)
-			if capacity > 0 {
+
+			// none of our specialized pods have room for another request
+			// check if we're already specializing a pod, and if we are, check if it has room for another request
+			numPodsSpecializing := funcSvcGroup.svcWaiting - funcSvcGroup.queue.Len()
+			maxRequestsForSpecializingPods := numPodsSpecializing * req.requestsPerPod
+
+			if maxRequestsForSpecializingPods-funcSvcGroup.svcWaiting > 0 {
```

Comment:
`maxRequestsForSpecializingPods - totalActiveRequests`

Reply:
I don't think `totalActiveRequests` is a substitute for `funcSvcGroup.svcWaiting`.

`totalActiveRequests` = number of requests being sent to a specialized pod
`funcSvcGroup.svcWaiting` = number of requests waiting for a pod to finish specializing